src/PersistentSubscriptionSettings.php

```diff
@@ -0,0 +1,215 @@
+<?php
+
+/**
+ * This file is part of prooph/event-store.
+ * (c) 2014-2018 prooph software GmbH <contact@prooph.de>
+ * (c) 2015-2018 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\EventStore;
+
+use Prooph\EventStore\Common\SystemConsumerStrategies;
+use Prooph\EventStore\Exception\InvalidArgumentException;
+
+class PersistentSubscriptionSettings
+{
+    /**
+     * Tells the subscription to resolve link events.
+     * @var bool
+     */
+    private $resolveLinkTos;
+    /**
+     * Start the subscription from the position-of the event in the stream.
+     * If the value is set to `-1` that the subscription should start from
+     * where the stream is when the subscription is first connected.
+     * @var int
+     */
+    private $startFrom;
+    /**
+     * Tells the backend to measure timings on the clients so statistics will contain histograms of them.
+     * @var bool
+     */
+    private $extraStatistics;
+    /**
+     * The amount of time the system should try to checkpoint after.
+     * @var int
+     */
+    private $checkPointAfterMilliseconds;
+    /**
+     * The size of the live buffer (in memory) before resorting to paging.
+     * @var int
+     */
+    private $liveBufferSize;
+    /**
+     * The size of the read batch when in paging mode.
+     * @var int
+     */
+    private $readBatchSize;
+    /**
+     * The number of messages that should be buffered when in paging mode.
+     * @var int
+     */
+    private $bufferSize;
+    /**
+     * The maximum number of messages not checkpointed before forcing a checkpoint.
+     * @var int
+     */
+    private $maxCheckPointCount;
+    /**
+     * Sets the number of times a message should be retried before considered a bad message.
+     * @var int
+     */
+    private $maxRetryCount;
+    /**
+     * Sets the maximum number of allowed subscribers.
+     * @var int
+     */
+    private $maxSubscriberCount;
+    /**
+     * Sets the timeout for a client before the message will be retried.
+     * @var int
+     */
+    private $messageTimeoutMilliseconds;
+    /**
+     * The minimum number of messages to write a checkpoint for.
+     * @var int
+     */
+    private $minCheckPointCount;
+    /** @var string */
+    private $namedConsumerStrategy;
+
+    private const INT_32_MAX = 2147483647;
+
+    public static function default(): self
+    {
+        return self::create()->build();
+    }
+
+    public static function create(): PersistentSubscriptionSettingsBuilder
+    {
+        return new PersistentSubscriptionSettingsBuilder(
```

Comment:
Instead of having builders, I think it would be easier to have empty constructor, put these default values inline in property declaration and provide `with*` methods.

Reply:
Don't like that idea.